app/persistence/transaction/transaction.go

```diff
@@ -72,23 +77,43 @@ func (tr *TransactionRepository) Create(ct *proto.CryptoTransferMessage) error {
 		EthAddress:    ct.EthAddress,
 		Amount:        ct.Amount,
 		Fee:           ct.Fee,
-		Status:        StatusPending,
+		Status:        StatusInitial,
 	}).Error
 }
 
-func (tr *TransactionRepository) UpdateStatusCancelled(txId string) error {
-	return tr.updateStatus(txId, StatusCancelled)
-}
-
 func (tr *TransactionRepository) UpdateStatusCompleted(txId string) error {
 	return tr.updateStatus(txId, StatusCompleted)
 }
 
-func (tr *TransactionRepository) UpdateStatusSubmitted(txId string, submissionTxId string, signature string) error {
+func (tr *TransactionRepository) UpdateStatusInsufficientFee(txId string) error {
+	return tr.updateStatus(txId, StatusInsufficientFee)
+}
+
+func (tr *TransactionRepository) UpdateStatusSignatureProvided(txId string) error {
+	return tr.updateStatus(txId, StatusSignatureProvided)
+}
+
+func (tr *TransactionRepository) UpdateStatusSignatureFailed(txId string) error {
+	return tr.updateStatus(txId, StatusSignatureFailed)
+}
+
+func (tr *TransactionRepository) UpdateStatusEthTxSubmitted(txId string, hash string) error {
+	return tr.dbClient.
+		Model(Transaction{}).
+		Where("transaction_id = ?", txId).
+		Updates(Transaction{Status: StatusSignatureSubmitted, EthHash: hash}).
```

Comment:
The status here should be `StatusEthTxSubmitted`

Reply:
noted.